src/allmydata/web/common.py

```diff
@@ -461,6 +461,99 @@ def _get_renderer(self, fmt):
 
 
 
+class MultiFormatResource(resource.Resource, object):
+    """
+    ``MultiFormatResource`` is a ``resource.Resource`` that can be rendered in
+    a number of different formats.
+
+    Rendered format is controlled by a query argument (given by
+    ``self.formatArgument``).  Different resources may support different
+    formats but ``json`` is a pretty common one.  ``html`` is the default
+    format if nothing else is given as the ``formatDefault``.
+    """
+    formatArgument = "t"
+    formatDefault = None
+
+    def render(self, req):
+        """
+        Dispatch to a renderer for a particular format, as selected by a query
+        argument.
+
+        A renderer for the format given by the query argument matching
+        ``formatArgument`` will be selected and invoked.  render_HTML will be
+        used as a default if no format is selected (either by query arguments
+        or by ``formatDefault``).
+
+        :return: The result of the selected renderer.
+        """
+        t = get_arg(req, self.formatArgument, self.formatDefault)
+        renderer = self._get_renderer(t)
+        return renderer(req)
+
+
+    def _get_renderer(self, fmt):
```

Comment:
Theoretically I think we're trying to follow pep8 for new Tahoe code (this looks like Twisted convention). There are still tons of places with *zero* spaces between functions :( but I think 1 blank for methods, 2 blanks for top-level functions / classes is what we're aiming at.

Reply:
Since there's some twisted mixed in with the lot.. you mean spacing-wise 1/2 blanks instead of the twisted 2/3? I haven't been sure about how to handle naming but it seems like `lower_with_underscores` for new code.

Reply:
Yes to both :)